inception/inception-search-core/src/main/java/de/tudarmstadt/ukp/inception/search/LayerStatistics.java

```diff
@@ -0,0 +1,157 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.search;
+
+public class LayerStatistics
+{
+    /** The stats to be calculated */
+    public static final String STATS = "n,sum,min,max,mean,median,standarddeviation";
+
+    private long total;
+    private long maximum;
+    private long minimum;
+    private double mean;
+    private double median;
+    private double standardDeviation;
+
+    private double maximumPerSentence;
+    private double minimumPerSentence;
+    private double meanPerSentence;
+    private double medianPerSentence;
+    private double standardDeviationPerSentence;
+
+    private long noOfDocuments;
+
+    public LayerStatistics()
+    {
+    }
+
+    public LayerStatistics(long aTotal, long aMaximum, long aMinimum, double aMean, double aMedian,
+            double aStandardDeviation, double aMaximumPerSentence, double aMinimumPerSentence,
+            double aMeanPerSentence, double aMedianPerSentence,
+            double aStandardDeviationPerSentence, long aNoOfDocuments)
+    {
+        total = aTotal;
+        maximum = aMaximum;
+        minimum = aMinimum;
+        mean = aMean;
+        median = aMedian;
+        standardDeviation = aStandardDeviation;
+
+        maximumPerSentence = aMaximumPerSentence;
+        minimumPerSentence = aMinimumPerSentence;
+        meanPerSentence = aMeanPerSentence;
+        medianPerSentence = aMedianPerSentence;
+        standardDeviationPerSentence = aStandardDeviationPerSentence;
+
+        noOfDocuments = aNoOfDocuments;
+    }
+
+    public boolean isLayerTrivial()
+    {
+        if (maximum == 0L) {
+            return true;
+        }
+        return false;
+    }
+
+    public long getTotal()
+    {
+        return total;
+    }
+
+    public long getMaximum()
+    {
+        return maximum;
+    }
+
+    public long getMinimum()
+    {
+        return minimum;
+    }
+
+    public double getMean()
+    {
+        return mean;
+    }
+
+    public double getMedian()
+    {
+        return median;
+    }
+
+    public double getStandardDeviation()
+    {
+        return standardDeviation;
+    }
+
+    public double getMaximumPerSentence()
+    {
+        return maximumPerSentence;
+    }
+
+    public double getMinimumPerSentence()
+    {
+        return minimumPerSentence;
+    }
+
+    public double getMeanPerSentence()
+    {
+        return meanPerSentence;
+    }
+
+    public double getMedianPerSentence()
+    {
+        return medianPerSentence;
+    }
+
+    public double getStandardDeviationPerSentence()
+    {
+        return standardDeviationPerSentence;
+    }
+
+    public long getNoOfDocuments()
+    {
+        return noOfDocuments;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (o == this) {
```

Comment:
I would use intellij to generate it with equalsbuilder instead and also would override hashCode at the same time.